PlagiarismPlugin.inc.php

```diff
@@ -43,215 +106,905 @@ public function getDescription() {
+	 * 		- Check if EULA is stamped to submitting user
+	 * 			- if not stamped, not allowed to submit at iThenticate
+	 *  - Traversing the submission files
+	 *  	- Create new submission at ithenticate's end for each submission file
+	 * 		- Upload the file for newly created submission uuid return back from ithenticate
+	 * 		- Stamp the retuning iThenticate submission id with submission file
+	 * 	- Return bool to indicate the status of process completion
+	 * 
+	 * @param string $hookName `submissionsubmitstep4form::execute`
+	 * @param array $args
+	 * 
+	 * @return bool
+	 */
+	public function submitForPlagiarismCheck($hookName, $args) {
+
+		// if the auto upload to ithenticate disable
+		// not going to upload files to iThenticate at submission time
+		if ($this->hasAutoSubmissionDisabled()) {
+			return false;
+		}
+
 		$request = Application::get()->getRequest();
+		$form =& $args[0]; /** @var SubmissionSubmitStep4Form $form */
+		$submission = $form->submission; /** @var Submission $submission */
 		$context = $request->getContext();
-		import('classes.notification.NotificationManager');
-		$notificationManager = new NotificationManager();
-		$roleDao = DAORegistry::getDAO('RoleDAO'); /* @var $roleDao RoleDAO */
-		// Get the managers.
-		$managers = $roleDao->getUsersByRoleId(ROLE_ID_MANAGER, $context->getId());
-		while ($manager = $managers->next()) {
-			$notificationManager->createTrivialNotification($manager->getId(), NOTIFICATION_TYPE_ERROR, array('contents' => __('plugins.generic.plagiarism.errorMessage', array('submissionId' => $submissionid, 'errorMessage' => $message))));
+		$user = $request->getUser();
+
+		if (!static::isRunningInTestMode() && !$this->isServiceAccessAvailable($context)) {
+			$this->sendErrorMessage("ithenticate service access not set for context id {$context->getId()}", $submission->getId());
+			return false;
+		}
+
+		$ithenticate = $this->initIthenticate(...$this->getServiceAccess($context)); /** @var \IThenticate $ithenticate */
+
+		// If no webhook previously registered for this Context, register it
+		if (!$context->getData('ithenticate_webhook_id')) {
+			$this->registerIthenticateWebhook($ithenticate, $context);
+		}
+
+		$ithenticate->setApplicableEulaVersion($submission->getData('ithenticate_eula_version'));
+
+		// Check EULA stamped to submission or submitter only if it is required
+		if ($this->getContextEulaDetails($context, 'require_eula') !== false) {
+			// not going to sent it for plagiarism check if EULA not stamped to submission or submitter
+			if (!$submission->getData('ithenticate_eula_version') || !$user->getData('ithenticateEulaVersion')) {
+				$this->sendErrorMessage('Unable to obtain the stamped EULA details to submission or submitter', $submission->getId());
+				return false;
+			}
+		}
+
+		/** @var DAOResultIterator $submissionFiles */
+		$submissionFiles = Services::get('submissionFile')->getMany([
+            'submissionIds' => [$submission->getId()],
```

Comment:
Your editor is mixing spaces and tabs here and elsewhere. This will go away in future branches once the PHP code formatter is added, but in the meantime, please make sure the formatting is done consistently.